packages/engine-server/src/markdown/remark/dendronPub.ts

```diff
@@ -491,5 +498,66 @@ function plugin(this: Unified.Processor, opts?: PluginOpts): Transformer {
   return transformer;
 }
 
+/** Generate elements to be included before and after the text of a wikilink. */
+function linkExtras({
+  note,
+  config,
+}: {
+  note?: NoteProps;
+  config: IntermediateDendronConfig;
+}): {
+  before: any[];
+  after: any[];
+} {
+  const before = [];
+  const after = [];
+
+  // For task notes, add the status, priority, due, and owner info
+  if (note && TaskNoteUtils.isTaskNote(note)) {
+    const taskConfig = ConfigUtils.getTask(config);
+    const status = TaskNoteUtils.getStatusSymbolRaw({
+      note,
+      taskConfig,
+    });
+    if (status) {
```

Comment:
since we already have css that styles checkboxes, instead of adding additional styles, can we convert them to the checkbox instead? 

when i add the following in the preview `- [ ] foo`, I get the following HTML:

```html
<ul class="contains-task-list">
    <li class="task-list-item">
    <input type="checkbox" disabled=""> foo bar</li>
</ul>
```

we can add the same html for the case when a task has no status or the status is marked as done. some caveats here:

- it appears that in gfm, checkboxes are only valid inside a bulleted list
    ```md
    [ ] this is not a task and will be rendered plainly
    - [ ] this is a task and will be rendered as a checkbox
    ```
    - if we wanted to follow this spec, we would need to inspect opening and closing `ul` tags (there's some extra complications here because users can intermix reegular bullet points with task bullet points)
- an 80/20 approach here could be to generate the following 
    ```html
    <div>
    <input type="checkbox" disabled=""> foo bar</li>
    </div>
    ```
    - this will cause the chexbox to force a line break and interacts well with lists

given this, I would propose the following:
- add the `div` based `checkbox` for tasks
- if a task is done, add a checked checkbox 
    - for now, consider a task checked if the status is set to `done` or `x` 
        - as a followup, for task configuration, we should be able to map `status` symbol as a `completion` symbol
- if a task has a status that is not `done`, add the status in brackets next to it `<input type="checkbox" disabled=""> [status] foo bar</li>`
- if a task has no status, add a empty disabled checkbox

Reply:
I went ahead and added some configuration for people to be able to define what a complete task is, and switched to using a checkbox instead.

- A status that is an empty string (or whitespace) will be displayed with an unchecked box
- A status that matches a state in the `taskCompleteStatus` will be displayed with a checked box
- Any other status will be displayed with an unchecked box, followed by the status in parenthesis.

I think this is a pretty clean look. The only caveat is that it looks like you can't really style disabled checkboxes, the workaround I've seen suggested is replicating a checkbox with HTML&CSS if you want disabled checkboxes with custom styles. That means the checkboxes have a dark color.

![2022-05-18-02-20-35-199828699](https://user-images.githubusercontent.com/1008124/168972038-59ef439e-0f5f-4a90-a0b1-9b42b6770ddd.png)
![2022-05-18-02-21-35-253612025](https://user-images.githubusercontent.com/1008124/168972041-cbca6873-a758-4d95-b3ee-60cb4617fb08.png)